extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -181,59 +181,67 @@ export class ProjectDashboard {
 			}).component();
 
 		dashboardData.forEach(info => {
-			const tableNameLabel = this.modelView!.modelBuilder.text()
-				.withProperties<azdata.TextComponentProperties>({ value: info.name, CSSStyles: { 'margin-block-start': '30px', 'margin-block-end': '0px' } })
-				.component();
-			tableContainer.addItem(tableNameLabel, { CSSStyles: { 'padding-left': '25px', 'padding-bottom': '20px', ...constants.cssStyles.title } });
-
-			const columns: azdata.DeclarativeTableColumn[] = [];
-			info.columns.forEach((column: IDashboardColumnInfo) => {
-				let col = {
-					displayName: column.displayName,
-					valueType: column.type === 'icon' ? azdata.DeclarativeDataType.component : azdata.DeclarativeDataType.string,
-					isReadOnly: true,
-					width: column.width,
-					headerCssStyles: {
-						'border': 'none',
-						...constants.cssStyles.tableHeader
-					},
-					rowCssStyles: {
-						...constants.cssStyles.tableRow
-					},
-				};
-				columns.push(col);
-			});
-
-			const data: azdata.DeclarativeTableCellValue[][] = [];
-			info.data.forEach(values => {
-				const columnValue: azdata.DeclarativeTableCellValue[] = [];
-				values.forEach(val => {
-					if (typeof val === 'string') {
-						columnValue.push({ value: val });
-					} else {
-						const iconComponent = this.modelView!.modelBuilder.image().withProperties<azdata.ImageComponentProperties>({
-							iconPath: val.icon,
-							width: '15px',
-							height: '15px',
-							iconHeight: '15px',
-							iconWidth: '15px'
-						}).component();
-						const stringComponent = this.modelView!.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
-							value: val.text,
-							CSSStyles: { 'margin-block-start': 'auto', 'block-size': 'auto', 'margin-block-end': '0px' }
-						}).component();
-
-						const columnData = this.modelView!.modelBuilder.flexContainer().withItems([iconComponent, stringComponent], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px' } }).withLayout({ flexFlow: 'row' }).component();
-						columnValue.push({ value: columnData });
-					}
+			if (info.data.length === 0) {
```

Comment:
does the dashboard get updated like if there's initially no data in the build table, but then a build is started? Or does the dashboard need to be closed and opened again?

Reply:
It needs to be opened again, doesn't get refreshed on the fly (nor does it have a Refresh button).

Reply:
can a refresh button be added? It would be confusing to the user how to refresh the dashboard and see their results and it's seems unintuitive to have to close and reopen the dashboard.

Reply:
I did think of this enhancement as well. Created https://github.com/microsoft/azuredatastudio/issues/15008 for now.

Reply:
Added a refresh button in latest iteration.

Reply:
assuming this is dont now - resolving for clarity